### ISSUE TITLE
SafeMode#value_for_name throws an exception if name is not found

### ISSUE THREAD

**Issue by ggrossetie** [asciidoctor/asciidoctor#3526]
But `SafeMode#name_for_value` returns `nil` if the value is not found. 
I think we should use the same behavior between the two methods (ie. either throws an exception or returns nil if the SafeMode is not found).

Since the Asciidoctor API does not extensively use exceptions, I think we should return `nil` if the name is not found:

```ruby
SafeMode.value_for_name 'server'   # returns 10 (SafeMode::SERVER)
SafeMode.value_for_name 'paranoid' # returns nil
```

**Comment by mojavelinux** [comment]
Note that I found this fallback for the safe option the document is being loaded:

https://github.com/asciidoctor/asciidoctor/blob/d26821f99ce762af106dc4e7fce3453c688047ce/lib/asciidoctor/document.rb#L326

However, I don't think the value for SECURE is what should be returned from the API, so I can change this to a falsy check instead to retain this fallback for the option value.

**Comment by mojavelinux** [comment]
Even though this is minor, it feels like it should be 2.1.x rather than 2.0.x.

**Comment by ggrossetie** [comment]
I agree it should be included in a 2.1.x release.

**Comment by mojavelinux** [comment]
Let's go!
